src/os/state/state.js

```diff
@@ -89,6 +89,24 @@ os.state.titleCompare = function(a, b) {
 };
 
 
+/**
+ * Compares states by whether or not they are supported
+ *
+ * @param {os.state.IState} a A state
+ * @param {os.state.IState} b Another state
+ * @return {number} The comparison
+ */
+os.state.supportCompare = function(a, b) {
+  if (a['supported'] && !b['supported']) {
```

Comment:
This function takes two `os.state.IState` instances, which do not define a `supported` property. You should add an `isSupported` method to the interface, and implement in `AbstractState` to expose the new property.